### DISCUSSION TITLE
How safe is git to use?

### DISCUSSION THREAD

**Discussion by bearcatsandor** [beetbox/beets#4732] [General]
Given that the periods between releases are vast, how safe is the git repo code?  Is it likely to eat music files or my painstakingly entered custom fields?  Yeah, I know. Have backups and all that, but I'm asking generally.

**Comment by RollingStar** [comment]
We had a big bug make it into 1.6.0 stable. Others can explain how bad the bug was, it looks bad to me. https://github.com/beetbox/beets/issues/4528

I believe the desire is for stable releases to be more stable than the dev/github version, but in practice, there doesn't seem to be much difference in stability or testing right now. We don't have a bunch of ninjas behind the scenes testing things. We have the automated tests we run, and the users who run tests, but it seems to be about it. The idea was to have more regular releases (every 3 months), but we don't have enough people to make that happen right now.

See the comments on a 'core team'. https://github.com/beetbox/beets/discussions/4558

Freezetag might be a good way to get tag backups without having to backup the entire files. https://github.com/x1ppy/freezetag

**Reply by daniele-athome** [reply]
Hi there! Is there a change log of what happened between 1.6.0 and master that is not the git history? :-)
Mainly I just wanted to know if there is any major thing that would make updating from 1.6.0 worth it.

Thanks

**Reply by bearcatsandor** [reply]
I would say that bug fix makes updating worth it.

**Reply by JOJ0** [reply]
@daniele-athome @bearcatsandor You'll always find the latest changelog alongside the latest/dev documentation: https://beets.readthedocs.io/en/latest/changelog.html

Usually submitting new fixes/features require contributors to include an entry in the changelog, so you can assume that it reflects the current state of development. HTH

**Comment by bearcatsandor** [comment]
Thank you for the reply. I got lost in my school work.  I appreciate the suggestion of freezetag, but all of my files are wavpack.  I'll install the git version  and see how well it works for me.

**Comment by JOJ0** [comment]
As for an answer to this initial question, I'd like to second what @RollingStar said about stable releases and dev versions. I think that usually you would exchange some old bugs with new ones, and additionally you would get some new features for free coming along with it :-)


 If you are happy with 1.6.0 and don't face any bug that is bothering you and don't want to use any new feature in that changelog, stay with stable.

If you are not afraid of reaching out for help and are a tinkering person in general (as a beets user you kind of always are ;-)) you should use the dev version. It might even help the project if you find oddities. Reporting old problems from 1.6.0 _sometimes_ doesn't help anyway, because they _often_ are reported already and might even have been addressed already.

My personal usage of beets even includes several features that not even made it into the master branch (open pull requests). I'm rarely facing crashes even with that bleeding edge approach. Since beets foundation is pretty solid in my opinion I'm not really afraid of killing any of my files. I do have backups (I keep my original unsorted music as-is and have set `copy: yes` when importing into beets). The biggest danger for killing my music files is "me, myself and I" hitting the return key at the wrong time.....;-)
